### PR TITLE
Fetch service network options

### PR DESCRIPTION
**Pull request by orlandohohmeier**
network-type

**AC**
* [x] Fetch the virtual network overlays and provide them as options in the deploy service modal 
* [x] Adjust the network type getter to properly set the default

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @MatApple

File: src/js/components/ServiceForm.js [hunk 1]
```diff
@@ -68,6 +70,26 @@ class ServiceForm extends SchemaForm {
...
+    let {networking} = this.multipleDefinition;
+    let virtualNetworks = VirtualNetworksStore.getOverlays()
+      .mapItems(function (overlay) {
+        let name = overlay.getName();
+
+        return {html: `Virtual Network: ${name}`, id: name}
+      }).getItems();
+
+    // networking.definition[0].options = [].concat(networkType.options,
+    // virtualNetworks);
```
Comment (pierlo-upitup) [inline src/js/components/ServiceForm.js]: shouldnt this be removed?

File: src/js/components/ServiceForm.js [hunk 2]
```diff
@@ -68,6 +70,26 @@ class ServiceForm extends SchemaForm {
...
+    networking.definition.forEach(function (definition) {
+      if (definition.name === 'networkType') {
+        definition.options = [].concat(networkType.options, virtualNetworks);
+      }
+      return definition;
```
Comment (pierlo-upitup) [inline src/js/components/ServiceForm.js]: who's receiving these?

File: src/js/utils/ServiceUtil.js [hunk 3]
```diff
@@ -324,7 +324,7 @@ const ServiceUtil = {
             definition.container.docker.network = 'BRIDGE';
           } else {
             definition.container.docker.network = 'USER';
-            definition.ipAddress = {networkName: 'd-overlay-1'};
+            definition.ipAddress = {networkName: networkType};
```
Comment (pierlo-upitup) [inline src/js/utils/ServiceUtil.js]: can't we stick to calling this `networkName`?

File: src/js/utils/__tests__/ServiceUtil-test.js [hunk 4]
```diff
@@ -322,9 +322,9 @@ describe('ServiceUtil', function () {
         it('adds the networkName field to the service', function () {
           let service = ServiceUtil.createServiceFromFormModel({
             containerSettings: { image: 'redis' },
-            networking: { networkType: 'user', ports: [{}] },
+            networking: { networkType: 'prod', ports: [{}] },
```
Comment (pierlo-upitup) [inline src/js/utils/__tests__/ServiceUtil-test.js]: wait – isnt `network type` still supposed to be USER?

**Comment by rcorral** [comment]
Code looks good to me 👍

File: src/js/utils/__tests__/ServiceUtil-test.js (hunk 4 above)
Comment (orlandohohmeier) [inline src/js/utils/__tests__/ServiceUtil-test.js]: I've to admit it's a little confusing and the prop name is anything but good (that's a different story and not part of this PR)... The network type is used to define whether service container network should be `HOST`, `BRIDGE`, or `USER` but it is also used to define the service `ipAddress.networkName`. Thus we assign the selected option to the `networkType` property and use that prop to determine the respective values. That's why we expect `service.ipAddress.networkName` to equal `networkType` and the reason why I adjusted the values.

File: src/js/utils/ServiceUtil.js (hunk 3 above)
Comment (orlandohohmeier) [inline src/js/utils/ServiceUtil.js]: See https://github.com/dcos/dcos-ui/pull/564#discussion_r68847355 for details.

**Pull request merged**